dev-docs/modules/jwplayerRtdProvider.md

```diff
@@ -0,0 +1,145 @@
+---
+layout: page_v2
+title: JW Player Real Time Data Provider
+display_name: JW Player video ad targeting
+description: makes JW Player's video ad targeting information accessible to Bid Adapters.
+page_type: module
+module_type: rtd
+module_code : jwplayer
+enable_download : true
+sidebarType : 1
+---
+
+# JW Player RTD Provider
+{:.no_toc}
+
+* TOC
+{:toc}
+
+## Overview
+
+The purpose of this Real Time Data Provider is to allow publishers to target against their JW Player media without 
+being forced to integrate with the Player Bidding product. This prebid module makes JW Player's video ad targeting information accessible 
+to Bid Adapters.
+
+## Implementation for Publishers:
+
+1) Compile the JW Player RTD Provider into your Prebid build:
+
+`gulp build --modules=jwplayerRtdProvider`
+
+2) Publishers must register JW Player as a Real Time Data provider by using `setConfig` to load a Prebid Config containing a `realTimeData.dataProviders` array:
+
+```javascript
+pbjs.setConfig({
+    ...,
+    realTimeData: {
+      auctionDelay: 100,
+      dataProviders: [{
+          name: "jwplayer",
+          waitForIt: true,
+          params: {
+            mediaIDs: ['abc', 'def', 'ghi', 'jkl']
+          }
+      }]
+    }
+});
+``` 
+
+3) In order to prefetch targeting information for certain media, include the media IDs in the `jwplayer` var and set `waitForIt` to `true` before calling `setConfig`:
+
+**Note:** `waitForIt` is required to ensure the auction waits for the prefetching of the relvant targeting information to complete. It signals to Prebid that you allow the module to delay the auction if necessary.
+
+**Note:** setting an `auctionDelay` in the `realTimeData` object is required to ensure the auction waits for prefetching to complete. The `auctionDelay` is the max time in ms that the auction will wait for the requested targeting information.
+
+**Config Syntax details:**
+
+{: .table .table-bordered .table-striped }
+| Name  |Type | Description   | Notes  |
+| :------------ | :------------ | :------------ |:------------ |
+| name | String | Real time data module name | Always 'jwplayer' |
+| waitForIt | Boolean | Required to ensure that the auction is delayed until prefetch is complete | Optional. Defaults to false |
+| params | Object | | |
+| params.mediaIDs | Array of Strings | Media Ids for prefetching | Optional |
+
+4) Include the content's media ID and/or the player's ID in the matching AdUnit's `fpd.context.data.jwTargeting` before calling `addAdUnits`:
+
+```javascript
+   const adUnit = {
+     code: '/19968336/prebid_native_example_1',
+     ...,
+     fpd: {
+       context: {
+         data: {
+           jwTargeting: {
```

Comment:
Why can't these come in on realTimeData.dataProviders[jwplayer].params?  is it because they can differ per adunit?

Reply:
@bretg correct